homeassistant/components/osoenergy/sensor.py

```diff
@@ -0,0 +1,179 @@
+        key="power_load",
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        value=lambda device: device.state,
+    ),
+    "tapping_capacity": OSOEnergySensorEntityDescription(
+        key="tapping_capacity",
+        translation_key="tapping_capacity",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        value=lambda device: device.state,
+    ),
+    "capacity_mixed_water_40": OSOEnergySensorEntityDescription(
+        key="capacity_mixed_water_40",
+        translation_key="capacity_mixed_water_40",
+        device_class=SensorDeviceClass.VOLUME,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        value=lambda device: device.state,
+    ),
+    "v40_min": OSOEnergySensorEntityDescription(
+        key="v40_min",
+        translation_key="v40_min",
+        device_class=SensorDeviceClass.VOLUME,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        value=lambda device: device.state,
+    ),
+    "v40_level_min": OSOEnergySensorEntityDescription(
+        key="v40_level_min",
+        translation_key="v40_level_min",
+        device_class=SensorDeviceClass.VOLUME,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        value=lambda device: device.state,
+    ),
+    "v40_level_max": OSOEnergySensorEntityDescription(
+        key="v40_level_max",
+        translation_key="v40_level_max",
+        device_class=SensorDeviceClass.VOLUME,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        value=lambda device: device.state,
+    ),
+    "volume": OSOEnergySensorEntityDescription(
+        key="volume",
+        device_class=SensorDeviceClass.VOLUME,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        value=lambda device: device.state,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up OSO Energy sensor."""
+    osoenergy = hass.data[DOMAIN][entry.entry_id]
+    devices = osoenergy.session.device_list.get("sensor")
+    entities = []
+    if devices:
+        for dev in devices:
+            sensor_type = dev.osoEnergyType.lower()
+            if sensor_type in SENSOR_TYPES:
+                entities.append(
+                    OSOEnergySensor(osoenergy, SENSOR_TYPES[sensor_type], dev)
+                )
+
+    async_add_entities(entities, True)
+
+
+class OSOEnergySensor(OSOEnergyEntity[OSOEnergySensorData], SensorEntity):
+    """OSO Energy Sensor Entity."""
+
+    entity_description: OSOEnergySensorEntityDescription
+
+    def __init__(
+        self,
+        instance: OSOEnergy,
+        description: OSOEnergySensorEntityDescription,
+        device: OSOEnergySensorData,
+    ) -> None:
+        """Initialize the OSO Energy sensor."""
+        self.device = None
```

Comment:
There's no need to set this to `None` since it's unconditionally set by `OSOEnergy.__init__`.